### PR TITLE
Enhacements.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Add `Breadcrumbs::class` widget.
- [x] Rework code `Dropdown::class`.
- [x] Add `DropdownTests`.
- [x] Rework code `Message::class`.
- [x] Rework code `NavBar::class`.
- [x] Rework code `Nav::class`, add support icon items.

### CODE REVIEW

File: src/Breadcrumbs.php [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+{
+    private bool $encodeLabels = true;
+    private array $homeLink = [];
+    private string $itemTemplate = "<li>{link}</li>\n";
+    private string $itemTemplateActive = "<li class=\"is-active\"><a aria-current=\"page\">{label}</li>\n";
+    private array $links = [];
+    private array $options = [];
+    private array $optionsItems = [];
+
+    protected function run(): string
```
Comment (romkatsu) [inline src/Breadcrumbs.php:31]: I suggest dividing it into methods for readability:

```php
    protected function run(): string
    {
        if (empty($this->links)) {
            return '';
        }

        $this->buildOptions();

        return Html::tag('nav', Html::tag('ul', implode('', $this->buildLinks()), $this->optionsItems), $this->options);
    }

    private function buildOptions(): void
    {
        if (!isset($this->options['id'])) {
            $this->options['id'] = $this->createId();
        }

        $this->options = $this->addOptions($this->options, 'breadcrumb');
        $this->options = array_merge(['aria-label' => 'breadcrumbs'], $this->options);
    }

    private function createId(): string
    {
        return "{$this->getId()}-breadcrumbs";
    }

    private function buildLinks(): array
    {
        $links = [$this->getHomeLink()];
        foreach ($this->links as $link) {
            $links[] = $this->createLink($link);
        }

        return $links;
    }

    private function getHomeLink(): string
    {
        if ($this->homeLink === array()) {
            return $this->createHomeLink();
        }

        return $this->renderItem($this->homeLink, $this->itemTemplate);
    }

    private function createHomeLink(): string
    {
        return $this->renderItem(
            [
                'label' => 'Home',
                'url' => '/',
            ],
            $this->itemTemplate
        );
    }

    private function createLink($link): string
    {
        if (!\is_array($link)) {
            $link = ['label' => $link];
        }

        return $this->renderItem($link, isset($link['url']) ? $this->itemTemplate : $this->itemTemplateActive);
    }

```

File: src/Breadcrumbs.php [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+ * ```php
+ * echo Breadcrumbs::widget()
+ *     ->links(['label' => !empty($this->params['breadcrumbs']) ? $this->params['breadcrumbs'] : []]);
+ * ```
+ */
+class Breadcrumbs extends Widget
+{
+    private bool $encodeLabels = true;
+    private array $homeLink = [];
+    private string $itemTemplate = "<li>{link}</li>\n";
```
Comment (romkatsu) [inline src/Breadcrumbs.php:25]: You might want to use a constant 

```php
private string $itemTemplate = "<li>{link}</li>" . PHP_EOL;
```

**Review by romkatsu — CHANGES_REQUESTED** [review]

File: src/Breadcrumbs.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Breadcrumbs.php:25]: If I use a constant you will not be able to configure the item template, the idea is that it be configurable, you can change the order or the tag.

File: src/Breadcrumbs.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Breadcrumbs.php:31]: implement `buildOptions()`, done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**